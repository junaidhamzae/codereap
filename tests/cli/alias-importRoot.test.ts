import path from 'node:path';
import fs from 'node:fs';
import os from 'node:os';
import { runCli } from '../helpers/cli';

function getRow(json: any[], endsWith: string) {
  return (json as any[]).find((r) => r.node.endsWith(endsWith));
}

describe('CLI alias/importRoot resolution', () => {
  it('resolves modules via tsconfig paths and importRoot', () => {
    const fixture = path.resolve(__dirname, '../fixtures/alias-importRoot');
    const tmp = fs.mkdtempSync(path.join(os.tmpdir(), 'cli-'));
    const outBase = path.join(tmp, 'report');
    const res = runCli(['--root', fixture, '--out', outBase, '--importRoot', './src']);
    expect(res.status).toBe(0);
    const rows = res.json as any[];
    // util and button should be live due to index.ts imports
    expect(getRow(rows, 'src/util.ts').orphan).toBe(false);
    expect(getRow(rows, 'src/components/Button/index.ts').orphan).toBe(false);
  });
});


