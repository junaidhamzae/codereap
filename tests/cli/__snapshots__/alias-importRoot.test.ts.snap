// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CLI alias/importRoot resolution resolves modules via tsconfig paths and importRoot 1`] = `
[
  {
    "exists": true,
    "in-degree": 0,
    "node": "package.json",
    "orphan": true,
    "size-bytes": 118,
    "symbols": {
      "exports": {
        "default": {
          "exists": false,
          "orphan": true,
          "referencedInFile": false,
        },
        "named": [],
        "reExports": [],
      },
      "imports": [],
    },
  },
  {
    "exists": true,
    "in-degree": 1,
    "node": "src/components/Button/index.ts",
    "orphan": false,
    "size-bytes": 27,
    "symbols": {
      "exports": {
        "default": {
          "exists": false,
          "orphan": true,
          "referencedInFile": false,
        },
        "named": [
          {
            "name": "Button",
            "orphan": true,
            "reexport": false,
            "referencedInFile": false,
          },
        ],
        "reExports": [],
      },
    },
  },
  {
    "exists": true,
    "in-degree": 0,
    "node": "src/index.ts",
    "orphan": false,
    "size-bytes": 53,
    "symbols": {
      "exports": {
        "default": {
          "exists": false,
          "orphan": true,
          "referencedInFile": false,
        },
        "named": [],
        "reExports": [],
      },
    },
  },
  {
    "exists": true,
    "in-degree": 1,
    "node": "src/util.ts",
    "orphan": false,
    "size-bytes": 22,
    "symbols": {
      "exports": {
        "default": {
          "exists": false,
          "orphan": true,
          "referencedInFile": false,
        },
        "named": [
          {
            "name": "u",
            "orphan": true,
            "reexport": false,
            "referencedInFile": false,
          },
        ],
        "reExports": [],
      },
    },
  },
  {
    "exists": true,
    "in-degree": 0,
    "node": "tsconfig.json",
    "orphan": true,
    "size-bytes": 143,
    "symbols": {
      "exports": {
        "default": {
          "exists": false,
          "orphan": true,
          "referencedInFile": false,
        },
        "named": [],
        "reExports": [],
      },
      "imports": [],
    },
  },
]
`;
